src/components/augmentedReality/AugmentedRealityView.js

```diff
@@ -125,40 +125,45 @@ const ViroSoundAnd3DObject = (item) => {
             <ViroVideo
               loop
               materials={['chromaKeyFilteredVideo']}
-              position={object?.mp4?.position}
-              rotation={object?.mp4?.rotation}
-              scale={object?.mp4?.scale}
-              source={{ uri: object?.mp4?.uri }}
+              position={object.mp4?.position}
+              rotation={object.mp4?.rotation}
+              scale={object.mp4?.scale}
+              source={{ uri: object.mp4?.uri }}
             />
           )}
 
           {!!object?.image && (
             <ViroImage
-              position={object?.image?.position}
-              rotation={object?.image?.rotation}
-              scale={object?.image?.scale}
-              source={{ uri: object?.image?.uri }}
+              position={object.image?.position}
+              rotation={object.image?.rotation}
+              scale={object.image?.scale}
+              source={{ uri: object.image?.uri }}
             />
           )}
         </>
       )}
 
-      <Viro3DObject
-        source={{ uri: object?.vrx?.uri }}
-        resources={object?.textures}
-        type="VRX"
-        position={object?.vrx?.position}
-        rotation={object?.vrx?.rotation}
-        scale={object?.vrx?.scale}
-        onLoadStart={() => setIsObjectLoading(true)}
-        onLoadEnd={() => setIsObjectLoading(false)}
-        onError={() => alert(texts.augmentedReality.arShowScreen.objectLoadErrorAlert)}
-        animation={{
-          loop: true,
-          name: object?.animationName,
-          run: isStartAnimationAndSound
-        }}
-      />
+      {!!object?.models &&
+        !!object?.textures &&
```

Comment:
`?.length` for both to ensure entries in these arrays?

Reply:
Yes, you are right 👍